include/swift/SIL/SILDebugInfoExpression.h

```diff
@@ -0,0 +1,121 @@
+//===--- SILDebugInfoExpression.h - DIExpression for SIL --------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+///
+//===----------------------------------------------------------------------===//
+#ifndef SWIFT_SIL_DEBUGINFOEXPRESSION_H
+#define SWIFT_SIL_DEBUGINFOEXPRESSION_H
+#include "swift/AST/Decl.h"
+#include "llvm/ADT/iterator_range.h"
+#include "llvm/ADT/Optional.h"
+#include "llvm/Support/raw_ostream.h"
+#include <unordered_map>
+
+namespace swift {
+class TailAllocatedDebugVariable;
+
+enum class SILDIExprOperator : unsigned {
+  INVALID = 0,
+  Fragment,
+  Dereference
+};
+
+struct SILDIExprOperand {
+  enum Kind {
+    K_Operator,
+    K_Decl
```

Comment:
Are we using a K_ prefix elsewhere? I would probably just drop it.

Reply:
Since there is also a member variable called "Operator" in this class, I'll rename `K_Operator` into `OperatorKind` to avoid ambiguous (so does `K_Decl`).